php-checks/src/main/java/org/sonar/php/checks/formatting/IndentationCheck.java

```diff
@@ -19,118 +19,174 @@
+  @Override
+  public void visitFunctionExpression(FunctionExpressionTree tree) {
+    super.visitFunctionExpression(tree);
+    if (check.isMethodArgumentsIndentation) {
+      checkArgumentsIndentation(
+        tree.parameters().parameters(),
+        tree.functionToken(),
+        startColumnForLine(tree.functionToken().line()),
+        (tree.parameters().closeParenthesisToken()), false);
+    }
+  }
+
+  @Override
+  public void visitClassDeclaration(ClassDeclarationTree tree) {
+    super.visitClassDeclaration(tree);
+    checkImplementListIndentation(tree);
+  }
+
+  private void checkImplementListIndentation(ClassDeclarationTree classTree) {
+    if (check.isInterfacesIndentation && classTree.is(Tree.Kind.CLASS_DECLARATION) && !classTree.superInterfaces().isEmpty()) {
+      List<Tree> interfaceList = ImmutableList.<Tree>copyOf(classTree.superInterfaces());
+      SyntaxToken classToken = classTree.classEntryTypeToken();
+      SyntaxToken lastInterfaceToken = ((PHPTree) Iterables.getLast(classTree.superInterfaces())).getFirstToken();
+      int expectedColumn = classToken.column() + PSR2_INDENTATION;
 
-      if (!isOnSameLine(node.getToken(), Iterables.getLast(interfaceList).getToken())) {
+      if (!TokenUtils.isOnSameLine(classToken, lastInterfaceToken)) {
+
+        if (!isCorrectlySplittedOnLines(classToken.line(), interfaceList)) {
+          check.reportIssue(String.format(INTERFACE_SPLIT_MESSAGE, classToken.line()), classTree.superInterfaces().get(0));
 
-        if (!isCorrectlySplittedOnLines(classDecLine, interfaceList)) {
-          formattingCheck.reportIssue("Either split this list into multiple lines or move it on the same line \"" + classDecLine + "\".",
-            interfaceList.get(0));
         } else if (!isCorrectlyIndented(expectedColumn, interfaceList)) {
-          formattingCheck.reportIssue("Align all interfaces in this list at column \"" + expectedColumn + "\".", interfaceList.get(0));
+          check.reportIssue(String.format(INTERFACE_INDENTATION, expectedColumn), classTree.superInterfaces().get(0));
         }
       }
     }
   }
 
-  private void checkArgumentsIndentation(FormattingStandardCheck formattingCheck, AstNode node, AstNodeType parameterNodeType) {
-    List<AstNode> arguments = node.getChildren(parameterNodeType);
-
+  private void checkArgumentsIndentation(SeparatedList arguments, SyntaxToken functionName, int baseColumn, SyntaxToken closeParenthesis, boolean isFunctionCall) {
     if (arguments.size() > 1) {
-      Token lastParam = Iterables.getLast(arguments).getLastToken();
-      AstNode methodName = node.getPreviousAstNode();
-      AstNode firstParam = arguments.get(0);
-      int expectedIndentationColumn = getLineStartingColumn(node.getParent()) + PSR2_INDENTATION;
-      int callingLine = methodName.getTokenLine();
+      SyntaxToken lastArg = ((PHPTree) Iterables.getLast(arguments)).getLastToken();
+      Tree firstArg = (Tree) arguments.get(0);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unnecessary cast to "Tree". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1905)